Documentation/proposals/202309-controller-id.md

```diff
@@ -0,0 +1,76 @@
+# Controller-ID Proposal
+
+* Owners:
+  * [danielmellado](https://github.com/danielmellado)
+* Related Tickets:
+  * [#4281](https://github.com/prometheus-operator/prometheus-operator/issues/4281)
+  * [#4498](https://github.com/prometheus-operator/prometheus-operator/pull/4498)
+* Other Docs:
+  * n/a
+
+# Introduction
+
+This proposal aims to implement a solution to support multiple cluster-level
+Prometheus instances running concurrently without conflicting over the same
+custom resources.
```

Comment:
This is probably obvious for all of us but the solution isn't going to be limited to the Prometheus resources and will also be available for Alertmanager and ThanosRuler (and any pod-based resource we may add in the future).

Reply:
I'll add a note about that, thanks!